packages/react-utils/src/useStableCallback.ts

```diff
@@ -0,0 +1,31 @@
+import { useCallback, useRef } from 'react'
+import { useIsomorphicEffect } from './useIsomorphicEffect'
+
+/**
+ * Returns a stable callback that is equal to the given callback.
+ * Useful for preventing unnecessary re-renders.
+ *
+ * @param callback - Function to stabilize
+ *
+ * @remarks
+ * As opposed to `useCallback` which recreates callback when dependencies change,
+ * `useStableCallback` does not recreate the callback itself, but rather the
+ * reference to it. You will always get the latest version of the callback.
+ *
+ * @example
+ * ```tsx
+ * const stableCallback = useStableCallback(() => getRandomValue())
+ * stableCallback()
+ * // always returns the latest value, but the reference to the
+ * // callback itself is stable and never changes
+ * ```
+ */
+export const useStableCallback = <A extends unknown[], T>(callback: (...args: A) => T) => {
```

Comment:
thought: Have we weighed in alternative names for this hook? For example `useMemoizedCallback`, `useStoredCallback` or similar? I found the naming of "stable" somewhat confusing. It got me wondering what it is that is unstable to start with. An unmemoized or recreated function isn't necessarily "unstable".

Reply:
Or `useReferencedCallback`?

When i talked to Dash about it, he showed me that the naming `useStableCallback` appears to be a convention:
- https://github.com/mariusbrataas/useStableCallback
- https://foundations.significa.co/hooks/use-stable-callback
- https://gist.github.com/Shrugsy/5e898173c965e7642db8927636bedf7a